datasource.go

```diff
@@ -3,600 +3,515 @@
+				req.Resolution = common.String(ts.Resolution)
+				req.StartTime = &common.SDKTime{Time: start}
+				req.EndTime = &common.SDKTime{Time: end}
+				if ts.ResourceGroup != "NoResourceGroup" {
+						req.ResourceGroup = common.String(ts.ResourceGroup)
+				}
+
+				reg := common.StringToRegion(ts.Region)
+				o.metricsClient.SetRegion(string(reg))
 
-		start = start.Truncate(time.Millisecond)
-		end = end.Truncate(time.Millisecond)
+				request := monitoring.SummarizeMetricsDataRequest{
+						CompartmentId:               common.String(ts.Compartment),
+						SummarizeMetricsDataDetails: req,
+				}
 
-		req := monitoring.SummarizeMetricsDataDetails{}
-		req.Query = common.String(ts.Query)
-		req.Namespace = common.String(ts.Namespace)
-		req.Resolution = common.String(ts.Resolution)
-		req.StartTime = &common.SDKTime{start}
-		req.EndTime = &common.SDKTime{end}
-		if ts.ResourceGroup != "NoResourceGroup" {
-			req.ResourceGroup = common.String(ts.ResourceGroup)
+				res, err := o.metricsClient.SummarizeMetricsData(ctx, request)
+				if err != nil {
+						return nil, errors.Wrap(err, fmt.Sprint(spew.Sdump(query), spew.Sdump(request), spew.Sdump(res)))
+				}
+				results = append(results, responseAndQuery{
+						res,
+						query,
+						err,
+				})
 		}
+		resp := backend.NewQueryDataResponse()
+		for _, q := range results {
+				respD := resp.Responses[q.query.RefID]
 
-		reg := common.StringToRegion(ts.Region)
-		o.metricsClient.SetRegion(string(reg))
+				if q.err != nil {
+						respD.Error = fmt.Errorf(q.err.Error())
+						continue
+				}
 
-		request := monitoring.SummarizeMetricsDataRequest{
-			CompartmentId:               common.String(ts.Compartment),
-			SummarizeMetricsDataDetails: req,
+				for _, item := range q.ociRes.Items {
+						name := *(item.Name)
+						dimensionKeys := make([]string, len(item.Dimensions))
+						i := 0
+
+						for key, dimension := range item.Dimensions {
+								if re.MatchString(key) {
+										name = fmt.Sprintf("%s, {%s}", name, dimension)
+								}
+								dimensionKeys[i] = key
+								i++
+						}
+
+						// if there isn't a human readable name fallback to resourceId
+						if name == *(item).Name {
+								var preDisplayName = ""
+								sort.Strings(dimensionKeys)
+								for _, dimensionKey := range dimensionKeys {
+										if preDisplayName == "" {
+												preDisplayName = item.Dimensions[dimensionKey]
+										} else {
+												preDisplayName = preDisplayName + ", " + item.Dimensions[dimensionKey]
+										}
+								}
+
+								name = fmt.Sprintf("%s, {%s}", name, preDisplayName)
+						}
+
+						frame := data.NewFrame(q.query.RefID,
+								data.NewField("Time", nil, []time.Time{}),
+								data.NewField("Value", nil, []float64{}).SetConfig(&data.FieldConfig{
+										DisplayNameFromDS: name,
+								}),
+						)
+
+						for _, metric := range item.AggregatedDatapoints {
+								frame.AppendRow(metric.Timestamp.UnixNano()/1000000, *(metric.Value))
```

Comment:
@athulmurali Could you try this instead?

```suggestion
								frame.AppendRow(metric.Timestamp.Time, *(metric.Value))
```